src/QirRuntime/test/unittests/QirRuntimeTests.cpp

```diff
@@ -841,4 +865,64 @@ TEST_CASE("Unpacking input tuples of nested callables (case1)", "[qir_support]")
     quantum__rt__qubit_release_array(controlsInner);
     quantum__rt__array_unreference(controlsInner);
     quantum__rt__qubit_release(target);
+}
+
+TEST_CASE("Allocation tracking for arrays", "[qir_support]")
+{
+    InitializeQirContext(nullptr /*don't need a simulator*/, true /*track allocations*/);
+
+    QirArray* bounce = quantum__rt__array_create_1d(1, 1);
+    quantum__rt__array_unreference(bounce);
+    CHECK_THROWS(quantum__rt__array_reference(bounce));
+
+    QirArray* releaseTwice = quantum__rt__array_create_1d(1, 1);
+    quantum__rt__array_unreference(releaseTwice);
+    CHECK_THROWS(quantum__rt__array_unreference(releaseTwice));
+
+    QirArray* maybeLeaked = quantum__rt__array_create_1d(1, 1);
+    CHECK_THROWS(ReleaseQirContext());
+
+    quantum__rt__array_unreference(maybeLeaked);
+    ReleaseQirContext();
+}
+
+TEST_CASE("Allocation tracking for tuples", "[qir_support]")
+{
+    InitializeQirContext(nullptr /*don't need a simulator*/, true /*track allocations*/);
+
+    PTuple bounce = quantum__rt__tuple_create(1);
+    quantum__rt__tuple_unreference(bounce);
+    CHECK_THROWS(quantum__rt__tuple_reference(bounce));
+
+    PTuple releaseTwice = quantum__rt__tuple_create(1);
+    quantum__rt__tuple_unreference(releaseTwice);
+    CHECK_THROWS(quantum__rt__tuple_unreference(releaseTwice));
+
+    PTuple maybeLeaked = quantum__rt__tuple_create(1);
+    CHECK_THROWS(ReleaseQirContext());
+
+    quantum__rt__tuple_unreference(maybeLeaked);
+    ReleaseQirContext();
+}
+
+void CallableEntry(PTuple, PTuple, PTuple) {}

```

Comment:
Minor:
**static** void CallableEntry(PTuple, PTuple, PTuple) {} ?

You may also want to consider making the function `CallableEntry()` a local function (move line 908 to between the lines 910 and 911). Such that if at some point in the future somebody decides to remove (or move/shift) the whole test case, then the function `CallableEntry()` will (re)moved together with the test case.
See "The static function of the local class" in [Nested (Local) Functions](https://github.com/kuzminrobin/code_review_notes/blob/master/cpp_design_bookmarks.md#nested-local-functions)

Reply:
Marked it static and renamed to `DummyCallableEntry`